sstables/sstables.cc

```diff
@@ -3112,18 +3112,16 @@ delete_atomically(std::vector<shared_sstable> ssts) {
             // Create temporary pending_delete log file.
             auto f = open_file_dma(tmp_pending_delete_log, oflags).get0();
             // Write all toc names into the log file.
-            file_output_stream_options options;
-            options.buffer_size = 4096;
-            auto w = file_writer::make(std::move(f), options, tmp_pending_delete_log).get0();
+            auto out = make_file_output_stream(std::move(f), 4096).get0();
```

Comment:
file_writer's destructor knows it's running in a thread and knows to close the file unless closed already. A generic output_stream does not, so defer() is needed here.